plugins/editor/js/editor.js

```diff
@@ -1682,16 +1682,54 @@
-                                        exec: function(composer, command) {
-                                            wysihtml5.commands.formatBlock.exec(composer, "formatBlock", "pre", "CodeBlock", REG_EXP);
-                                            if ($(composer.element.lastChild).hasClass('CodeBlock')) {
-                                                composer.selection.setAfter(composer.element.lastChild);
-                                                composer.commands.exec("insertHTML", "<p><br></p>");
+                                        exec: function(composer) {
+                                            var inCodeBlock = this.state(composer);
+                                            if (inCodeBlock) {
+                                                // caret is already within a <pre><code>...</code></pre>
+                                                composer.selection.executeAndRestore(function() {
+                                                    var pre = wysihtml5.dom.getParentElement(
+                                                        composer.selection.getSelectedNode(),
+                                                        {
+                                                            className: "CodeBlock",
+                                                            classRegExp: /CodeBlock/g,
+                                                            nodeName: "PRE"
+                                                        },
+                                                        3
+                                                    );
+                                                    var code = pre.querySelector("code");
+
+                                                    wysihtml5.dom.replaceWithChildNodes(pre);
+                                                    if (code) {
+                                                        wysihtml5.dom.replaceWithChildNodes(code);
+                                                    }
+                                                });
+                                            } else {
+                                                // Wrap in <pre><code>...</code></pre>
+                                                var range = composer.selection.getRange();
+                                                var selectedNodes = range.extractContents();
+                                                var pre = composer.doc.createElement("pre");
+                                                var code = composer.doc.createElement("code");
+
+                                                pre.className = "CodeBlock";
+                                                pre.appendChild(code);
+                                                code.appendChild(selectedNodes);
+                                                range.insertNode(pre);
+                                                composer.selection.selectNode(pre);
```

Comment:
When creating a new code block, the cursor is outside of the code block. Changing this to `composer.selection.selectNode(code);` fixes this.